src/elm/aggregate.coffee

```diff
@@ -160,6 +160,43 @@ module.exports.StdDev = class StdDev extends AggregateExpression
     pop_dev = Math.sqrt pop_var
     {standard_variance: std_var, population_variance: pop_var, standard_deviation: std_dev, population_deviation: pop_dev}
 
+module.exports.Product = class Product extends AggregateExpression
+  constructor:(json) ->
+    super
+
+  exec: (ctx) ->
+    arg = @source.execute(ctx)
```

Comment:
I think `list` or `listOfValues` would be more descriptive than `arg` here.  Simlarly below, `product` instead of `value` for:
`[value, filtered] = productValue(arg)`
(And same suggestions for GeometricMean variables)